File: DemoServer/PythonDemo/demo_methods/Basic/StaticQuery.py
from demo.holder import DocumentStoreHolder
from datetime import timedelta
from demo.entities import Order
from demo.indexes import CostlyOrders
from pyravendb.tools.utils import Utils


class StaticQuery(object):
    @staticmethod
    def static_query(high_price=500, delay_days=35):
        with DocumentStoreHolder.get_store().open_session() as session:
            result = list(
                session.query(index_name=CostlyOrders.__name__, object_type=Order).
                    where_greater_than("Price", high_price).
                    and_also().
                    where_greater_than("Delay", Utils.timedelta_tick(timedelta(days=delay_days))))

            return result[0] if result else None

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/basic/StaticQuery.java
package net.ravendb.demo.basic;

import net.ravendb.client.documents.session.IDocumentQuery;
import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Order;
import net.ravendb.demo.indexes.CostlyOrders;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.time.Duration;
import java.util.List;

@Controller
public class StaticQuery {
    @GetMapping("/basic/staticQuery")
    public List<Order> staticQuery(
            @RequestParam(value = "highPrice", defaultValue = "500") String highPriceStr,
            @RequestParam(value = "delayDays", defaultValue = "35") String delayDaysStr) {

        int highPrice = Integer.parseInt(highPriceStr);
        int delayDeys = Integer.parseInt(delayDaysStr);

        //GetImportantOrdersWithIssues - High Price Orders and Delayed Orders
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {

            IDocumentQuery<Order> query = session.query(CostlyOrders.Result.class, CostlyOrders.class)
                    .whereGreaterThan("Price", highPrice)
                    .whereGreaterThan("Delay", Duration.ofDays(delayDeys).toNanos() / 100)
                    .ofType(Order.class);

            List<Order> problematicOrders = query.toList();

            return problematicOrders;
        }
    }
}
